crates/ruff/resources/test/fixtures/pylint/unspecified_encoding.py

```diff
@@ -0,0 +1,163 @@
+"""Warnings for using open() without specifying an encoding"""
+import dataclasses
+import io
+import locale
+from pathlib import Path
+from typing import Optional
+
+FILENAME = "foo.bar"
+open(FILENAME, "w", encoding="utf-8")
+open(FILENAME, "wb")
+open(FILENAME, "w+b")
+open(FILENAME)  # [unspecified-encoding]
```

Comment:
One issue here is that this rule is in conflict with `UP015`, which removes redundant open modes (so, e.g., it changes `open("foo", "r")` to `open("foo")`). We'd have to mark these rules as incompatible in the `INCOMPATIBLE_CODES` slice in `crates/ruff/src/registry.rs`.